planet/clients/orders.py

```diff
@@ -313,6 +338,26 @@ async def download_order(self,
                                       progress_bar=progress_bar)
             for location in locations
         ]
+        if checksum:
+            # Checksum Implementation
+            # Get manifest filepath
+            manifest_json = ' '.join([x for x in filenames if x.endswith('manifest.json')])
+            # Save each filename and respective hash in a dict as a key-value pair
+            with open(manifest_json, 'rb') as manifest:
+                manifest_data = json.load(manifest)
+                file_key_pairs = {}
+                for json_entry in manifest_data['files']:
+                    file_name = json_entry['path'].split('/')[-1]
+                    origin_hash = json_entry['digests'][checksum.lower()]
+                    file_key_pairs[file_name] = origin_hash
+            # For each file (not including manifest json file), calculate hash on its contents
+            filenames_loop = [x for x in filenames if not x.endswith('manifest.json')]
+            # filenames_loop = ' '.join([x for x in filenames if not x.endswith('manifest.json')])
+            for filename in filenames_loop:
+                downloaded_file_name = filename.split('/')[-1]
+                self.calculate_checksum(origin_hash= file_key_pairs[downloaded_file_name],
```

Comment:
On line 352 you store the file name and MD5 key for `manifest.json` in `file_key_pairs`. And on line 357 `downloaded_file_name` is the name of the a file that's not `manifest.json`, so as of right now `file_key_pairs[downloaded_file_name]` won't work. However, a simple solution would be to change from `file_key_pairs[downloaded_file_name]` to `file_key_pairs['manifest.json']`, so you're really just trying to pull the key from the `manifest.json`.

Reply:
`manifest.json` has several keys inside it for different files- I don't actually want to use the `manifest.json` itself (it has no keys tied to it) which is why it's left out.